### ISSUE TITLE
Geth Fatal: Could not open database

### ISSUE THREAD

**Issue by swlfigo** [bnb-chain/bsc#1996]
Using Geth v1.3.1
Following Official Documentation:
https://docs.bnbchain.org/docs/validator/fullnode/

I download Geth fast node Snapshot from 48Club:
https://github.com/48Club/bsc-snapshots#geth-fast-node

First I create datadir using this command:
`geth --datadir ./node --db.engine=pebble init ./genesis.json`

then I replace donwloaded snapshot chaindata to replace folder in node's same place
`./node/geth/chaindata`

then I run Geth with this command:
`geth --config ./config.toml --datadir ./node  --cache 8000 --rpc.allow-unprotected-txs --txlookuplimit=0 --syncmode=full --tries-verify-mode=none --pruneancient=true --history.transactions=0 --db.engine=pebble`

It shows error:
```
database=/node/geth/chaindata cache=3.12GiB 
databaseVersion: 8 (0x8)
headBlockHash: 0x753f3b10dc5400e0dd27e9470e5148ff557d3101a46a82724586838971fc1fcd
headFastBlockHash: x753f3b10dc5400e0dd27e9470e5148ff557d3101a46a82724586838971fc1fco
headHeaderHash: 0x753f3b10dc5400e0dd27e9470e5148ff557d3101a46a82724586838971fc1fcd
lastPivotNumber: <nil>
len(snapshotSyncStatus): o bytes
snapshotDisabled: false
snapshotJournal: 6355917 bytes
snapshotRecoveryNumber: <nil>
snapshotRoot: 0x660887bd6cddadb76041e13a5d21933a18c494104993ed2a1666de3cf63b2836
txIndexTail:  (x0)
fastTxLookupLimit: <nil>
Fatal: Could not open database: open /node/geth/chaindata/ancient/chain/bodies.cidx: no such file or directorv
```

I have tried backup files in ancient/chain after first initial with genesis.json, which it included database needed files, after move 48club snapshot chaindata to datadir, I replace original ancient/chain data in it. And Geth finally run and sync successfully

After successfully run geth , the file in node/geth/chaindata/ancient will lost and can't run next time

Is any wrong step I maked?

**Comment by RenRick** [comment]
”After successfully run geth , the file in node/geth/chaindata/ancient will lost and can't run next time“
Could you pls share the log? Or is it the same error with that you attached above? thanks

**Comment by joeylichang** [comment]
Step3：delete  --pruneancient=true flag try again.
`geth --config ./config.toml --datadir ./node  --cache 8000 --rpc.allow-unprotected-txs --txlookuplimit=0 --syncmode=full --tries-verify-mode=none --history.transactions=0 --db.engine=pebble`

note: don't replace snapshot's ancient/chain data.

**Comment by swlfigo** [comment]
> Step3：delete --pruneancient=true flag try again. `geth --config ./config.toml --datadir ./node --cache 8000 --rpc.allow-unprotected-txs --txlookuplimit=0 --syncmode=full --tries-verify-mode=none --history.transactions=0 --db.engine=pebble`
> 
> note: don't replace snapshot's ancient/chain data.

The snapshot downloaded from 48club, there is no xxxxx.cidx files in snapshot/geth/chaindata/ancient/  folder, if I don't replace folder created with genesis.json , Geth will show can't open DB which can't find these files.

**Comment by swlfigo** [comment]
> ”After successfully run geth , the file in node/geth/chaindata/ancient will lost and can't run next time“ Could you pls share the log? Or is it the same error with that you attached above? thanks

@RenRick same error, After First Success run, all file (eg. xxxx.cidx ) all missed in ancient folder.

**Comment by weiihann** [comment]
This should be due to the pruneancient bug, which has been fixed in this PR https://github.com/bnb-chain/bsc/pull/1986.

**Comment by flywukong** [comment]
Compiled Try with v1.3.1 + this PR https://github.com/bnb-chain/bsc/pull/1986,  and try run with the same snapshot and the commands as the issue descriped,  it works .  Pls notice  it is needed  to replace original ancient/chain data now.

**Comment by swlfigo** [comment]
> This should be due to the pruneancient bug, which has been fixed in this PR #1986.

problem is fixed with this PR, Thx!

**Comment by swlfigo** [comment]
> problem is fixed with this PR, Thx!

problem is fixed with this PR, Thx!
